src/test/java/com/danit/controllers/employee/GymControllerTest.java

```diff
@@ -0,0 +1,164 @@
+        .perform(get(url).headers(header))
+        .andExpect(status().isOk());
+  }
+
+  @Test
+  public void getAllGyms() throws Exception {
+    int currentQuant = gymService.getGymQuant();
+    HttpHeaders header = testUtils.getHeader(template, UserRolesEnum.USER);
+    this.mockMvc.perform(post(url).headers(header)
+        .contentType("application/json")
+        .content("{\n"
+            + "    \"name\": \"Test gym\",\n"
+            + "    \"description\": \"Test gym\"\n"
+            + "  }"))
+        .andExpect(status().isOk());
+
+    mockMvc.perform(get(url).headers(header))
+        .andExpect(content().contentType(MediaType.APPLICATION_JSON_UTF8))
+        .andExpect(jsonPath("$", hasSize(currentQuant + 1)));
+  }
+
+  @Test
+  public void updateGymTest() throws Exception {
+    HttpHeaders header = testUtils.getHeader(template, UserRolesEnum.USER);
+
+    String responseJson = this.mockMvc.perform(post(url).headers(header)
+        .contentType("application/json")
+        .content("{\n"
+            + "    \"name\": \"Test gym\",\n"
+            + "    \"description\": \"Test gym\"\n"
+            + "  }"))
+        .andExpect(status().isOk())
+        .andReturn().getResponse().getContentAsString();
+
+    ObjectMapper mapper = new ObjectMapper();
+    Gym actualObj = mapper.readValue(responseJson, new TypeReference<Gym>() {
+    });
+    long createdId = actualObj.getId();
+    System.out.println(actualObj);
+
+    responseJson = mockMvc.perform(put(url+"/" + createdId).headers(header)
+        .contentType("application/json")
+        .content("{\n"
+            + "    \"id\": " + createdId + ", \n"
+            + "    \"name\": \"TestGym2\"\n"
+            + "  }"))
+        .andExpect(status().isOk())
+        .andReturn().getResponse().getContentAsString();
+
+    actualObj = mapper.readValue(responseJson, new TypeReference<Gym>() {
+    });
+    System.out.println(actualObj);
+    assertEquals("TestGym2", actualObj.getName());
+
+  }
+
+  @Test
+  public void createGymTest() throws Exception {
+    int currentQuant = gymService.getGymQuant();
+    HttpHeaders header = testUtils.getHeader(template, UserRolesEnum.USER);
+    this.mockMvc.perform(post(url).headers(header)
+        .contentType("application/json")
+        .content("{\n"
+            + "    \"name\": \"Test gym\",\n"
+            + "    \"description\": \"Test gym\"\n"
+            + "  }"))
+        .andExpect(status().isOk());
+
+    mockMvc.perform(get(url).headers(header))
+        .andExpect(content().contentType(MediaType.APPLICATION_JSON_UTF8))
+        .andExpect(jsonPath("$", hasSize(currentQuant + 1)));
+  }
+
+  @Test
+  public void expect500WhenNoDataFoundService() throws Exception {
+    HttpHeaders header = testUtils.getHeader(template, UserRolesEnum.USER);
+
+    mockMvc.perform(get(url+"/0").headers(header))
+        .andExpect(status().isNotFound());
+  }
+
+  @Test
+  public void expect500WhenDeleteNonexistentService() throws Exception {
```

Comment:
Ideally DELETE non-existing object should return 404 too.

Reply:
mockMvc return 500